core/aws_ddk_core/resources/_sfn.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Dict, Optional
+
+import aws_cdk.aws_stepfunctions as sfn
+from aws_ddk_core.config import Config
+from aws_ddk_core.resources.commons import BaseSchema, Duration
+from constructs import Construct
+from marshmallow import ValidationError, fields
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+class StateMachineSchema(BaseSchema):
+    """DDK Step Functions State Machine Marshmallow schema."""
+
+    tracing_enabled = fields.Bool()
+    timeout = Duration()
+    state_machine_type = fields.Method(deserialize="load_state_machine_type")
+
+    def load_state_machine_type(self, value: str) -> sfn.StateMachineType:
+        state_machine_types: Dict[str, sfn.StateMachineType] = sfn.StateMachineType._member_map_
+        try:
+            return state_machine_types[value.upper()]
+        except KeyError as error:
+            raise ValidationError(
+                f"`state_machine_type` value must be one of {state_machine_types.values()}."
+            ) from error
+
+
+class StepFunctionsFactory:
+    """
+    Class factory create and configure AWS Step Functions Service DDK resources, including State Machines.
+    """
+
+    @staticmethod
+    def state_machine(
+        scope: Construct,
+        environment_id: str,
+        id: str,
+        state_machine_name: Optional[str] = None,
+        state_machine_type: Optional[sfn.StateMachineType] = None,
+        timeout: Optional[Duration] = None,
+        tracing_enabled: Optional[bool] = None,
+        **state_machine_props: Any,
+    ) -> sfn.StateMachine:
+        """
+        Create and configure state machine.
+
+        This construct allows to configure parameters of the queue using ddk.json configuration file
```

Comment:
Minor - copy/paste `parameters of the queue`

Reply:
Thanks